PaymentSystem/Server/Application/Transfers/Commands/MakeTransferCommand.cs

```diff
@@ -12,27 +12,58 @@
 
 namespace PaymentSystem.Server.Application.Transfers.Commands
 {
-    public class MakeTransferCommand : IRequest<BoolResult>
+    public class MakeTransferCommand : IRequest<MakeTransferResult>
     {
         public string UserId { get; set; }
         public TransferDto Data { get; set; }
     }
 
-    public class MakeTransferCommandHandler : IRequestHandler<MakeTransferCommand, BoolResult>
+    public class MakeTransferResult 
+    {
+        public bool IsSuccessful { get; set; }
+        public string SuccessMessage { get; set; }
+        public string FailureReason { get; set; }
+
+        public static MakeTransferResult ReturnSuccess()
+        {
+            return new MakeTransferResult { IsSuccessful = true };
+        }
+
+        public static MakeTransferResult ReturnSuccess(string successMessage)
+        {
+            return new MakeTransferResult {
+                IsSuccessful = true,
+                SuccessMessage = successMessage
+            };
+        }
+
+
+        public static MakeTransferResult ReturnFailure(string failureReason)
+        {
+            return new MakeTransferResult
+            {
+                IsSuccessful = false,
+                FailureReason = failureReason
+            };
+        }
+
+    }
+
+    public class MakeTransferCommandHandler : IRequestHandler<MakeTransferCommand, MakeTransferResult>
     {
         private readonly ApplicationDbContext context;
         public MakeTransferCommandHandler(ApplicationDbContext context)
         {
             this.context = context;
         }
 
-        public async Task<BoolResult> Handle(MakeTransferCommand command, CancellationToken cancellationToken)
+        public async Task<MakeTransferResult> Handle(MakeTransferCommand command, CancellationToken cancellationToken)
         {
             var user = await context.Users.Include(x => x.Wallets).FirstOrDefaultAsync(x => x.Id == command.UserId);
 
             if (!user.Wallets.Any(x => x.Currency == command.Data.Currency))
             {
-                return BoolResult.ReturnFailure();
+                return MakeTransferResult.ReturnFailure("MISSING_SOURCE_WALLET");
```

Comment:
Should we have constants for this somewhere?

Reply:
Yep, I've added internal response message enums for each result, thus keeping the code more readable, thanks!